sashimi-webapp/src/logic/filemanager/data/folder.js

```diff
@@ -116,19 +123,24 @@ Folder.prototype.remove = function remove() {
  * @return {Promise}
  */
 Folder.prototype.rename = function rename(newFolderName) {
-  return new Promise((resolve, reject) => {
-    if (this.id === ROOT_FOLDER_ID) {
-      reject(ERROR_RENAME_ROOTFOLDER);
-    }
-    if (hasSameFolderName.call(this, newFolderName)) {
-      reject(ERROR_RENAME_FOLDER);
-    }
-    resolve();
-  })
-  .then(() => storage.renameFolder(this.id, newFolderName))
-  .then(() => {
-    const oldFolderName = this.name;
-    this.name = newFolderName;
-    this.path = this.path.replace(oldFolderName, newFolderName);
-  });
+  if (!(this instanceof Folder)) {
+    throw new Error(ERROR_NOT_FOLDER_INSTANCE);
+  }
+
+  if (this.name === newFolderName) {
+    return new Promise((resolve, reject) => resolve());
+  /* }  else if (newFolderName.match(ILLEGAL_CHARACTERS)) {
+    return new Promise((resolve, reject) => reject(ERROR_CONTAIN_ILLEGAL_CHARACTERS));*/
+  } else if (this.id === ROOT_FOLDER_ID) {
+    return new Promise((resolve, reject) => reject(ERROR_RENAME_ROOTFOLDER));
```

Comment:
If you need to return a rejection only, you can use
`Promise.reject(reason)`

It is also a good practice to pass in an Error object for the `reject()` e.g.
```javascript
Promise.reject(new Error(ERROR_RENAME_ROOTFOLDER));
```